xarray/tests/test_dask.py

```diff
@@ -917,3 +918,176 @@ def test_dask_layers_and_dependencies():
     assert set(x.foo.__dask_graph__().dependencies).issuperset(
         ds.__dask_graph__().dependencies
     )
+
+
+def make_da():
+    da = xr.DataArray(
+        np.ones((10, 20)),
+        dims=["x", "y"],
+        coords={"x": np.arange(10), "y": np.arange(100, 120)},
+        name="a",
+    ).chunk({"x": 4, "y": 5})
+    da.coords["c2"] = 0.5
+    return da
+
+
+def make_ds():
+    map_ds = xr.Dataset()
+    map_ds["a"] = make_da()
+    map_ds["b"] = map_ds.a + 50
+    map_ds["c"] = map_ds.x + 20
+    map_ds = map_ds.chunk({"x": 4, "y": 5})
+    map_ds["d"] = ("z", [1, 1, 1, 1])
+    map_ds["z"] = [0, 1, 2, 3]
+    map_ds["e"] = map_ds.x + map_ds.y + map_ds.z
+    map_ds.coords["c1"] = 0.5
+    map_ds.coords["cx"] = ("x", np.arange(len(map_da.x)))
+    map_ds.coords["cxy"] = (("x", "y"), map_da.x * map_da.y)
+    map_ds.coords["cxy"] = map_ds.cxy.chunk({"y": 10})
+    map_ds.attrs["test"] = "test"
+    map_ds["xx"] = (map_ds["a"] * map_ds.y).chunk({"y": 20})
+
+    return map_ds
+
+
+# work around mypy error
+# xarray/tests/test_dask.py:888: error: Dict entry 0 has incompatible type "str": "int"; expected "Hashable": "Union[None, Number, Tuple[Number, ...]]"
+map_da = make_da()
+map_ds = make_ds()
+
+
+def test_unify_chunks():
+    with raises_regex(ValueError, "inconsistent chunks"):
+        map_ds.chunks
+
+    expected = {"x": (4, 4, 2), "y": (5, 5, 5, 5)}
+    actual = map_ds.unify_chunks().chunks
+    assert expected == actual
+
+
+# TODO: DataArray.chunks is not a dict but Dataset.chunks is!
+def assert_chunks_equal(a, b):
```

Comment:
I think this should be moved to xarray.testing.

Reply:
Yes I agree, forgot to add a comment asking for opinions about that.